File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        dig = 0
        pro = 1
        for i in str(n):
            dig += int(i)
            pro *= int(i)

        if(n % (dig + pro) == 0):
            return True

        return False

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int dig = 0;
        int pro = 1;
        int temp = n;
        while(temp != 0){
            int digit = temp % 10;

            dig += digit;
            pro *= digit;
            temp /= 10; 
        }
        int ans = pro + dig;
        if(n % ans == 0){
            return true;
        }
        return false;
    }
};